app/static/src/app/store/basic/actions.ts

```diff
@@ -5,19 +5,30 @@ import { BasicConfig } from "../../types/responseTypes";
 import { BasicMutation } from "./mutations";
 import { ErrorsMutation } from "../errors/mutations";
 import { AppStateMutation } from "../AppState";
+import { CodeMutation } from "../code/mutations";
+import { ModelAction } from "../model/actions";
 
 export enum BasicAction {
     FetchConfig = "FetchConfig"
 }
 
 export const actions: ActionTree<BasicState, BasicState> = {
     async [BasicAction.FetchConfig](context, appName) {
-        const { commit } = context;
+        const { commit, state, dispatch } = context;
         commit(AppStateMutation.SetAppName, appName);
-        await api(context)
+        const response = await api(context)
             .freezeResponse()
             .withSuccess(BasicMutation.SetConfig)
             .withError(`errors/${ErrorsMutation.AddError}` as ErrorsMutation, true)
             .get<BasicConfig>(`/config/${appName}`);
+
+        if (response) {
+            commit(`code/${CodeMutation.SetCode}`, state.config!.defaultCode, { root: true });
+
+            if (state.code.code.length) {
```

Comment:
looks like you could reduce the `state.code.code.length` to `state.code.length`, the type was unnecessarily extended.

Reply:
The first `code` is the module, which is going to have other stuff in it too (like validation errors) eventually, whereas the second `code` is the actual current code. Agreed it is a bit ugly to have those named the same. Would it be better if the property was `currentCode` (as opposed to `defaultCode` which we have in the config module).